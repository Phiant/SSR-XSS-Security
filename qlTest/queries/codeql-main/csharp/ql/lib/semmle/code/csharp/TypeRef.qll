/**
 * INTERNAL: Do not use.
 *
 * Provides support for type-references.
 */

import csharp

/** A typeref is a reference to a type in some assembly. */
private class TypeRef extends @typeref {
  /** Gets the name of type being referenced. */
  string getName() { typerefs(this, result) }

  /** Gets a textual representation of this type reference. */
  string toString() { result = this.getName() }

  /** Gets the type being referenced. */
  Type getReferencedType() {
    typeref_type(this, result)
    or
    not typeref_type(this, _) and
    result.(UnknownType).isCanonical()
  }
}

/**
 * INTERNAL: Do not use.
 *
 * Gets a type reference for a given type `type`.
 * This is used for extensionals that can be supplied
 * as either type references or types.
 */
TypeRef getTypeRef(Type type) { result.getReferencedType() = type }
