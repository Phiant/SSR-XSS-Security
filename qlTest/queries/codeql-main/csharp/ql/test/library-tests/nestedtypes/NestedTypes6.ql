/**
 * @name Test for nested types
 */

import csharp

from UnboundGenericClass o, UnboundGenericClass i
where
  o.hasFullyQualifiedName("NestedTypes", "Outer`1") and
  i.hasFullyQualifiedName("NestedTypes", "Outer`1+Inner`1") and
  i = o.getANestedType()
select o, i
