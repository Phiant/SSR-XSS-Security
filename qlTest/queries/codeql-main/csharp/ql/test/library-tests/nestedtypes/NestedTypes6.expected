| nestedtypes.cs:53:11:53:18 | Outer`1 | nestedtypes.cs:56:15:56:22 | Inner`1 |
